config.py

```diff
@@ -60,8 +60,11 @@ class ChannelConfig:
             increment: int
             cap: int
             message_rate: float
+            apply_to_threads: bool = True
             last_messages: int = 0
-        channels = {}
+            threads_last_messages: dict[str, int] = {}
+            
+        channels: dict[str, ChannelConfig] = {}
```

Comment:
```suggestion
            threads_last_messages: Dict[str, int] =field(default_factory=dict)
            
        channels: Dict[str, ChannelConfig] = {}
```

For version reasons we need to use Dict not dict. Also you need to give it field with default rather than empty

Reply:
We also need to import Dict & field

Reply:
What python version is being used to run the bot?